front_end/models/react_native/ReactDevToolsBindingsModel.ts

```diff
@@ -54,10 +56,44 @@ export class ReactDevToolsBindingsModel extends SDK.SDKModel.SDKModel {
     }
 
     if (parsedMessage) {
-      this.dispatchMessageToDomainEventListeners(parsedMessage.domain, parsedMessage.message);
+      const domainName = parsedMessage.domain;
+
+      // Messages are flushed out when execution context is created and dispatcher is initialized
+      if (this.fuseboxDispatcherIsInitialized && this.isDomainMessagesQueueEmpty(domainName)) {
```

Comment:
> In other words, can we make sure the messages are always already flushed when fuseboxDispatcherIsInitialized is true? That way we can just check for the boolean flag without checking the queue

Theoretically, we can't guarantee this. On practice, based on the fact that JS is single-threaded, yes, we can make this assumption. What could potentially happen is that Backend may send a message to Frontend, while Frontend is flushing out the messages from the queue. But because JS is single threaded, the event handler on the Frontend side (Chrome DevTools UI) is only going to handle this event (Backend -> Frontend message) once its done with flushing out messages.

```
Flush Message 1: Frontend -> Backend
Flush Message 2: Frontend -> Backend
...
Backend -> Frontend
...
Flush Message N: Frontend -> Backend
```

Soooo, I can remove this check, but add an assertion which will guarantee that queue is empty, what do you think?